src/application/downloader.py

```
@@ -0,0 +1,213 @@
+                "country_code": job.country_code,
+                "current_location_id": location.id
+            }
+            await self.storage.save_checkpoint(job.id, checkpoint)
+            
+            self.metrics.increment_counter(
+                "locations_completed",
+                tags={"country": job.country_code}
+            )
+            
+        except Exception as e:
+            logger.error(f"Failed to download location {location.id}: {e}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)